add preset options to dropdown task

mcbouslog opened the pull request:
The following preset options are available to be applied when building a dropdown task's options:

-Countries
-US states
-Canada provinces
-Mexico states
-months
-range of numbers (could be used for years, days, etc.)

Code is on a branch in my forked PFE repo, located [here](https://github.com/mcbouslog/Panoptes-Front-End/tree/dropdown-presets). 
Branch is staged [here](https://dropdown-presets.pfe-preview.zooniverse.org/). Enable the experimental dropdown task in a project to test adding preset dropdown options.

I'm anticipating some edits, please let me know what you think, no recommendation too small (like renaming `data` folder to `presets`?). Thanks!

closes Issue #2507

brian-c commented on app/classifier/tasks/dropdown/dropdown-dialog.cjsx:
```diff
@@ -182,6 +230,34 @@ DropdownDialog = React.createClass
 
+      <div>
+        <h2 className="form-label">Presets</h2>
+        <select ref="preset" onChange={@handlePresetChange}>
```
We're already setting state for `numberRange`, why not just keep the value of this select as state, since we eventually change it directly?

brian-c commented on app/classifier/tasks/dropdown/dropdown-dialog.cjsx:
```diff
@@ -182,6 +230,34 @@ DropdownDialog = React.createClass
+        {if @state.numberRange
+          <div>
+            <label>Top number:
+              <input type="number" ref="topNumber" min="0" max="999999" />
```
Why set a min/max on these?

brian-c commented on app/classifier/tasks/dropdown/dropdown-dialog.cjsx:
```diff
@@ -97,6 +105,46 @@ DropdownDialog = React.createClass
+    bottomNumber = @refs.bottomNumber.value
+
+    options = []
+    for num in [topNumber..bottomNumber]
```
The `value` of an HTML input is always a string, so this is a range between two strings. It works because JavaScript coerces the values into numbers, but it's prettyyy weird. Maybe `parseFloat` those values first.

brian-c commented:
Yeah I think "presets" makes more sense. Also style-wise, file names are dash-separated, and variable names starting with a capital letter are reserved for classes.

I've left some comments in-line. In general, modifying the DOM directly (i.e. `@refs.someInput.value = ''`) should be avoided in favor of setting state and rendering with the right value.

mcbouslog commented:
thanks Brian!

1. i think i properly changed the preset select to use state instead of @refs, please take a look and confirm. i'll add a comment on line 89 regarding a different, but similar scenario.
2. removed min/max, no good reason to have
3. parseFloat appears to work great
4. renamed folders, files and variables

mcbouslog commented on app/classifier/tasks/dropdown/dropdown-dialog.cjsx:
```diff
@@ -77,7 +85,7 @@ DropdownDialog = React.createClass
   onClickAddOption: ->
-    @addOption(@refs.optionInput.value)
+    @addOption @refs.optionInput.value
     @refs.optionInput.value = ''
```
should this also be updated to use state instead of refs? i'm fairly certain i know how to, but in my head it looks like more code so i'm not sure if this is a scenario where editing the ref/DOM directly is ok? editing state instead of the ref/DOM directly makes total sense though.

brian-c replied:
Sorry for the delay. I think this is fine here. Manipulating the DOM directly in a React component is very unusual, I'd avoid it in the future.
